### PR TITLE
Configuration for tox (pytest, isort, flake8) and travis

### PR DESCRIPTION
**Pull request by trojkat**
Related issue: #3 

The basic setup for running tests and linters.

Usage example:
```
╰(clove)─λ tox                                                                                                                         git-daemon-1 | 0 < 16:10:13
GLOB sdist-make: /home/trojkat/workspace/clove/setup.py
python inst-nodeps: /home/trojkat/workspace/clove/.tox/dist/clove-0.0.1.zip
python installed: attrs==17.4.0,bitcoinlib==0.3.29,certifi==2017.11.5,chardet==3.0.4,clove==0.0.1,ecdsa==0.13,flake8==3.5.0,idna==2.6,isort==4.2.15,mccabe==0.6.1,pbkdf2==1.3,pluggy==0.6.0,py==1.5.2,pycodestyle==2.3.1,pycrypto==2.6.1,pyflakes==1.6.0,pytest==3.3.2,requests==2.18.4,scrypt==0.8.0,six==1.11.0,SQLAlchemy==1.2.1,urllib3==1.22
python runtests: PYTHONHASHSEED='4031248711'
python runtests: commands[0] | pip install -e .[testing]
Obtaining file:///home/trojkat/workspace/clove
Requirement already satisfied: bitcoinlib==0.3.29 in ./.tox/python/lib/python3.6/site-packages (from clove==0.0.1)
Requirement already satisfied: pytest==3.3.2 in ./.tox/python/lib/python3.6/site-packages (from clove==0.0.1)
Requirement already satisfied: isort==4.2.15 in ./.tox/python/lib/python3.6/site-packages (from clove==0.0.1)
Requirement already satisfied: flake8==3.5.0 in ./.tox/python/lib/python3.6/site-packages (from clove==0.0.1)
Requirement already satisfied: scrypt in ./.tox/python/lib/python3.6/site-packages (from bitcoinlib==0.3.29->clove==0.0.1)
Requirement already satisfied: sqlalchemy in ./.tox/python/lib/python3.6/site-packages (from bitcoinlib==0.3.29->clove==0.0.1)
Requirement already satisfied: pbkdf2 in ./.tox/python/lib/python3.6/site-packages (from bitcoinlib==0.3.29->clove==0.0.1)
Requirement already satisfied: requests in ./.tox/python/lib/python3.6/site-packages (from bitcoinlib==0.3.29->clove==0.0.1)
Requirement already satisfied: ecdsa in ./.tox/python/lib/python3.6/site-packages (from bitcoinlib==0.3.29->clove==0.0.1)
Requirement already satisfied: pycrypto in ./.tox/python/lib/python3.6/site-packages (from bitcoinlib==0.3.29->clove==0.0.1)
Requirement already satisfied: pluggy<0.7,>=0.5 in ./.tox/python/lib/python3.6/site-packages (from pytest==3.3.2->clove==0.0.1)
Requirement already satisfied: setuptools in ./.tox/python/lib/python3.6/site-packages (from pytest==3.3.2->clove==0.0.1)
Requirement already satisfied: py>=1.5.0 in ./.tox/python/lib/python3.6/site-packages (from pytest==3.3.2->clove==0.0.1)
Requirement already satisfied: attrs>=17.2.0 in ./.tox/python/lib/python3.6/site-packages (from pytest==3.3.2->clove==0.0.1)
Requirement already satisfied: six>=1.10.0 in ./.tox/python/lib/python3.6/site-packages (from pytest==3.3.2->clove==0.0.1)
Requirement already satisfied: mccabe<0.7.0,>=0.6.0 in ./.tox/python/lib/python3.6/site-packages (from flake8==3.5.0->clove==0.0.1)
Requirement already satisfied: pycodestyle<2.4.0,>=2.0.0 in ./.tox/python/lib/python3.6/site-packages (from flake8==3.5.0->clove==0.0.1)
Requirement already satisfied: pyflakes<1.7.0,>=1.5.0 in ./.tox/python/lib/python3.6/site-packages (from flake8==3.5.0->clove==0.0.1)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in ./.tox/python/lib/python3.6/site-packages (from requests->bitcoinlib==0.3.29->clove==0.0.1)
Requirement already satisfied: urllib3<1.23,>=1.21.1 in ./.tox/python/lib/python3.6/site-packages (from requests->bitcoinlib==0.3.29->clove==0.0.1)
Requirement already satisfied: idna<2.7,>=2.5 in ./.tox/python/lib/python3.6/site-packages (from requests->bitcoinlib==0.3.29->clove==0.0.1)
Requirement already satisfied: certifi>=2017.4.17 in ./.tox/python/lib/python3.6/site-packages (from requests->bitcoinlib==0.3.29->clove==0.0.1)
Installing collected packages: clove
  Found existing installation: clove 0.0.1
    Uninstalling clove-0.0.1:
      Successfully uninstalled clove-0.0.1
  Running setup.py develop for clove
Successfully installed clove
python runtests: commands[1] | py.test
====================================================================== test session starts =======================================================================
platform linux -- Python 3.6.4, pytest-3.3.2, py-1.5.2, pluggy-0.6.0
rootdir: /home/trojkat/workspace/clove, inifile:
collected 1 item

tests/test_sample.py .                                                                                                                                     [100%]

==================================================================== 1 passed in 0.01 seconds ====================================================================
python runtests: commands[2] | isort -rc -c -df
Skipped 4 files
python runtests: commands[3] | flake8
____________________________________________________________________________ summary _____________________________________________________________________________
  python: commands succeeded
  congratulations :)

```

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -2,4 +2,32 @@
...
+force_sort_within_sections=true
+forced_separate=clove
+length_sort=False
+
+[flake8]
+exclude =
+    .git
+    .tox
+    docs
+max-line-length = 120
```
Comment (pafff) [inline setup.cfg:23]: Shouldn't it match isort `line_length`?

Comment (trojkat) [reply setup.cfg:23]: Oh, right! 👍

**Review by pafff — APPROVED** [review]

**Comment by trojkat** [comment]
Added `tox` to testing requirements.

**Review by szerepak — APPROVED** [review]
:tada: well done

**Pull request merged**